modules/consensus/consistency.go

```diff
@@ -2,281 +2,331 @@ package consensus
+	err := tx.ForEach(func(name []byte, b *bolt.Bucket) error {
+		// If the bucket is not a delayed siacoin output bucket or a file
+		// contract expiration bucket, skip.
+		if !strings.HasPrefix(string(name), string(prefixDSCO)) {
+			return nil
+		}
 
-	// Get the set of delayed siacoin outputs, sorted by maturity height then
-	// sorted by id and add them.
-	for i := cs.height() + 1; i <= cs.height()+types.MaturityDelay; i++ {
-		var delayedSiacoinOutputs crypto.HashSlice
-		if cs.db.inDelayedSiacoinOutputs(i) {
-			cs.db.forEachDelayedSiacoinOutputsHeight(i, func(id types.SiacoinOutputID, output types.SiacoinOutput) {
-				delayedSiacoinOutputs = append(delayedSiacoinOutputs, crypto.Hash(id))
-			})
+		// Add the bucket to the dscoTracker.
+		var height types.BlockHeight
+		err := encoding.Unmarshal(name[len(prefixDSCO):], &height)
+		if build.DEBUG && err != nil {
+			panic(err)
 		}
-		sort.Sort(delayedSiacoinOutputs)
-		for _, delayedSiacoinOutputID := range delayedSiacoinOutputs {
-			delayedSiacoinOutput := cs.db.getDelayedSiacoinOutputs(i, types.SiacoinOutputID(delayedSiacoinOutputID))
-			tree.PushObject(delayedSiacoinOutput)
-			tree.PushObject(delayedSiacoinOutputID)
+		_, exists := dscoTracker[height]
+		if exists {
+			return errors.New("repeat dsco map")
 		}
-	}
+		dscoTracker[height] = struct{}{}
+
+		var total types.Currency
+		err = b.ForEach(func(idBytes, delayedOutput []byte) error {
+			// Check that the output id has not appeared in another dsco.
+			var id types.SiacoinOutputID
+			copy(id[:], idBytes)
+			_, exists := idMap[id]
+			if exists {
+				return errors.New("repeat delayed siacoin output")
+			}
+			idMap[id] = struct{}{}
 
-	// Add the siafund pool
-	var siafundPool types.Currency
-	err := cs.db.Update(func(tx *bolt.Tx) error {
-		siafundPool = getSiafundPool(tx)
+			// Sum the funds in the bucket.
+			var sco types.SiacoinOutput
+			err := encoding.Unmarshal(delayedOutput, &sco)
+			if build.DEBUG && err != nil {
+				panic(err)
+			}
+			total = total.Add(sco.Value)
+			return nil
+		})
+		if err != nil {
+			return err
+		}
+
+		// Check that the minimum value has been achieved - the coinbase from
+		// an earlier block is guaranteed to be in the bucket.
+		minimumValue := types.CalculateCoinbase(height - types.MaturityDelay)
+		if total.Cmp(minimumValue) < 0 {
+			return errors.New("total number of coins in the delayed output bucket is incorrect")
+		}
 		return nil
 	})
 	if err != nil {
 		panic(err)
 	}
-	tree.PushObject(siafundPool)
 
-	return tree.Root()
+	// Check that all of the correct heights are represented.
+	currentHeight := blockHeight(tx)
+	expectedBuckets := 0
+	for i := currentHeight + 1; i <= currentHeight+types.MaturityDelay; i++ {
+		if i < types.MaturityDelay {
+			continue
+		}
+		_, exists := dscoTracker[i]
+		if !exists {
+			panic("missing a dsco bucket")
+		}
+		expectedBuckets++
+	}
+	if len(dscoTracker) != expectedBuckets {
+		panic("too many dsco buckets")
```

Comment:
not as worried about it in any TestXXX function, or testXXX function either. The consistency stuff should probably be qualified though.